trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmAccessionImporter.php

```diff
@@ -0,0 +1,885 @@
+    }
+    $organism_array = chado_get_organism_id_from_scientific_name($organism_name);
+
+    if (!$organism_array) {
+      $this->logger->error("Could not find an organism \"@organism_name\" in the database.", ['@organism_name' => $organism_name]);
+      $this->error_tracker = TRUE;
+      return false;
+    }
+    // We also want to check if we were given only one value back, as there is
+    // potential to retrieve multiple organism IDs
+    if (is_array($organism_array) && (count($organism_array) > 1)) {
+      $this->logger->error("Found more than one organism ID for \"@organism_name\" when only 1 was expected.", ['@organism_name' => $organism_name]);
+      $this->error_tracker = TRUE;
+      return false;
+    }
+
+    return $organism_array[0];
+  }
+
+  /**
+   * Checks if a stock exists in Chado and if not, inserts it and returns the primary
+   * key in the stock table. If the stock already exists, logs an error
+   *
+   * @param string $germplasm_name
+   *   The name of the germplasm.
+   * @param string $accession_number
+   *   A unique identifier for the germplasm accession.
+   * @param int $organism_id
+   *   The primary key of the stock's organism in the organism table
+   * @return int|false
+   *   The value of the primary key for the stock record in Chado. If the stock already
+   *   exists and does not match the accession number or type, or it cannot be inserted,
+   *   then FALSE is returned.
+   */
+  public function getStockID($germplasm_name, $accession_number, $organism_id) {
+
+    $accession_type_id = $this->getCVterm('accession');
+
+    // First query the stock table:
+    // 1. Using a regular condition to ensure the organism_id is a match
+    // 2. Create an OR condition group to look for records that match germplasm name OR 
+    //    the uniquename. Since the unique constraint is organism_id/uniquename/type_id,
+    //    we have to make sure this combo doesn't already exist with a different germplasm 
+    //    name.
+    $query = $this->connection->select('1:stock', 's')
+      ->fields('s', ['stock_id', 'name', 'uniquename', 'type_id'])
+      ->condition('s.organism_id', $organism_id, '=');
+
+    $orGroup = $query->orConditionGroup()
+      ->condition('s.name', $germplasm_name, '=')
+      ->condition('s.uniquename', $accession_number, '=');
+    
+    // Now add the OR condition group to the query
+    $query->condition($orGroup);
+    $record = $query->execute()->fetchAll();
+
+    // We may have retrieved 1+ records that share the germplasm name and/or 1+ records that
+    // share the accession_number. In this case, throw an error since there's no way to
+    // enter a new record with a unique organism_id/uniquename/type_id combo in this scenario
+    if (sizeof($record) >= 2) {
+      $this->logger->error("Found more than one stock ID for \"@germplasm_name\" and/or \"@accession\".", ['@germplasm_name' => $germplasm_name, '@accession' => $accession_number]);
```

Comment:
It would be really helpful here if you reported the stocks you did select. Something like 'Found more than one stock ID for "@germplasm_name" and/or "@accession". Specifically, name1 (accession1), name2 (accession2), name3 (accession3) already exist in the stock table.'

Reply:
Addressed in commit 5028b0fc4bcbb3ffae4acf5246f455e0bc9d777c using the format: **name1 (uniquename=accession1; stock_id=stockid1)**